Autopilot/Src/MPXV7002DP.cpp

```diff
@@ -0,0 +1,92 @@
+#include "airspeed.hpp"
+#include "adc.h"
+#include <stddef.h>
+
+#include <cmath>
+
+#define VS 5
+#define DENSITY_AIR 1.2754
+#define ADC_RESOLUTION 4096
+#define KPA_TO_PA 1000
+#define BASELINE 2048 // 2047.5
+
+uint16_t MPXV7002DP::GetAverage() {
+	uint16_t avg = 0;
+	for(int i = 0; i < 10; i++) {
+		HAL_ADC_PollForConversion(&hadc3, 50);
+		uint16_t val = HAL_ADC_GetValue(&hadc3);
+		avg += val;
+	}
+	avg = avg/10;
+	return avg;
+}
+
+MPXV7002DP::MPXV7002DP() {
+    //constructor
+    HAL_StatusTypeDef status = HAL_ADC_Start(&hadc3);
+    if(status != HAL_OK) {
+        sensorStatus = 1;
+    }
+
+    this->CalcOffset();
+}
+
+MPXV7002DP* MPXV7002DP::s_Instance = NULL;
+
+MPXV7002DP* MPXV7002DP::GetInstance() {
+    if(!s_Instance) {
+        s_Instance = new MPXV7002DP;
+    }
+    return s_Instance;
+}
```

Comment:
If you are interested in a more C++ style of doing singletons, you can refer to Anthony's imu driver in devel